### PR TITLE
Blog App: Views

### PR DESCRIPTION
**Pull request by kkolade**
# In this PR, I:

- Implement the design from the sneak peek wireframes.
   -  No styling added yet
- I use methods that I created in the 'Blog App - processing data in models'

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Post < ApplicationRecord
 
   after_save :update_post_counter
 
-  def user_posts_counter=(count)
+  def update_posts_counter=(count)
     author.update_attribute 'posts_counter', count
   end
```
Comment (franclobo) [inline app/models/post.rb:14]: - [ ] Please 🙏🏽 correct the update_post_counter method to be able to save the count of the post after saving it.

```ruby
after_save :update_post_counter

  def update_post_counter
    author.update_attribute('post_counter', author.posts.count)
  end
```

File: app/views/users/index.html.erb
Comment (franclobo) [inline app/views/users/index.html.erb:1]: - [ ] The user index page should match the wireframe, please 🙏🏽 edit the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image

File: app/views/users/show.html.erb
Comment (franclobo) [inline app/views/users/show.html.erb:1]: - [ ] The user show page should match the wireframe, please 🙏🏽 edit the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

File: app/views/posts/index.html.erb
Comment (franclobo) [inline app/views/posts/index.html.erb:1]: - [ ] The post index page should match the wireframe, please 🙏🏽 edit the styles to be similar to the wireframe 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

File: app/views/posts/show.html.erb
Comment (franclobo) [inline app/views/posts/show.html.erb:1]: - [ ] The post show page should match the wireframe, please 🙏🏽 edit the styles to be similar to the wireframe, kindly review and correct the `post_comment_counter` and `post_like_counter` methods to be able to see the comments and likes 👇🏽 

| Wireframe | Your design |
|---|---|
| image | image |

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @kkolade, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly redo the migrations in the correct order to avoid rails being aborted, first create the Users, and after that the Post, then the comments and likes.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 2]
```diff
@@ -9,11 +9,11 @@ class Post < ApplicationRecord
...
-    author.update_attribute 'posts_counter', count
+  def update_post_counter
+    author.update_attribute('post_counter', author.posts.count)
   end
 
   def last_five_comments
-    comments.last(5)
+    comments.order(created_at: :desc).limit(5)
   end
 end
```
Comment (FrederickMih) [inline app/models/post.rb:19]: - [Optional] In the posts table, you got:

```
    t.integer "comment_counter", default: 0
    t.integer "like_counter", default: 0
```

- But you changed the name of these two columns when validating them in the Post model as seen below:

```
  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }

```

- This creates the issue on the console below when setting up your app locally:
Screenshot from 2023-07-01 18-53-52

- Kindly fix the minor issue before merging your PR
- You change the column name from `text` to `body` I'll kindly suggest you pay attention always to the ERD proposed to you upfront

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @kkolade, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 

:tada: :champagne:  :tada: :champagne: 

## Highlights :green_circle: 

- The design of the app views is in line with the proposed wireframes :+1: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kkolade** [comment]
Hi @FrederickMih Thanks for the PR approval, please can you provide more clarification on the optional change requirement

**Pull request merged**